File: 1653/1653.cpp
class Solution {
  public:
      int minimumDeletions(string s) {
          int bs = 0;
          int deletions = 0;
          for(char& c : s) {
              if (c=='a'){deletions = min(deletions+1, bs);}
              else{bs+=1;}
          }
          return deletions;
      }
};

File: 1653/1653.java
class Solution {
    public int minimumDeletions(String s) {
        int bCount = 0;
        int deletionCount = 0;
        for (int i = 0; i < s.length(); i++){
            if (s.charAt(i)=='a') deletionCount = Math.min(deletionCount+1, bCount);
            else bCount++;
        }
        return deletionCount;
    }
}
